File: server/src/main/java/com/example/Finoana/Entity/Product.java
package com.example.Finoana.Entity;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper=false)
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@Entity
public class Product {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private LocalDateTime createdAt;
	private LocalDateTime updatedAt;
	
	private String name;
	private double price;
	private String image;
	private boolean availability;
	
	@Enumerated(EnumType.STRING)
	private Category category;
	
	private List<String> ingredients;
	
	@ManyToOne
	@JoinColumn(name = "chef_id")
	private Chef chef;
}

File: client/src/app/admin/models/product.ts
import { Category, Chef } from "./chef";

export interface ProductRequest {
    id : number;
    name : string;
    price : number;
    image : string;
    availability : boolean;
    category : Category | string;
    ingredients : string[];
    idChef : number | null;
}

export interface ProductResponse {
    id : number;
    name : string;
    price : number;
    image : string;
    availability : boolean;
    category : Category | string ;
    ingredients : string[],
    chef : Chef;
}